filemanager/config.go

```diff
@@ -0,0 +1,71 @@
+package filemanager
+
+import (
+	"os"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+)
+
+type ProviderConfigOpts struct {
+	Provider           string
+	Bucket             string
+	Prefix             string
+	Config             *config.Config
+	ExternalIDSupplier func() string
+}
+
+func GetProviderConfigFromEnv(opts ProviderConfigOpts) map[string]interface{} {
+	if opts.Config == nil {
+		opts.Config = config.Default
+	}
+	if opts.ExternalIDSupplier == nil {
+		opts.ExternalIDSupplier = func() string { return "" }
+	}
+	config := opts.Config
```

Comment:
[optional] to avoid name clash with the package name

```suggestion
	conf := opts.Config
```